authentication/auth_method_client_secret.go

```diff
@@ -1,25 +1,33 @@
 package authentication
 
 import (
+	"context"
 	"fmt"
+	"strings"
 
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/hashicorp/go-multierror"
+	"github.com/manicminer/hamilton/auth"
+	"github.com/manicminer/hamilton/environments"
 )
 
 type servicePrincipalClientSecretAuth struct {
+	ctx            context.Context
```

Comment:
rather than doing this could we add `ctx` to the method parameters below? as a general rule you shouldn't store the context in Go afaik

Reply:
Makes sense, will look at updating these methods accordingly